xplique/metrics/fidelity.py

```diff
@@ -198,6 +197,35 @@ def evaluate(self,
             Metric score, area over the deletion (lower is better) or insertion (higher is
             better) curve.
         """
+        scores_dict = self.detailed_evaluate(explanations)
+
+        # compute auc using trapezoidal rule (the steps are equally distributed)
+        np_scores = np.array(list(scores_dict.values()))
+        auc = np.mean(np_scores[:-1] + np_scores[1:]) * 0.5
+
+        return auc
+
+    def detailed_evaluate(self,
+                          explanations: Union[tf.Tensor, np.ndarray]) -> Dict[int, float]:
+        """
+        Evaluate model performance for successive perturbations of an input.
+        Used to compute causal score.
+
+        Allow visualize the deletion and insertion curves.
+
+        Parameters
+        ----------
+        explanations
+            Explanation for the inputs, labels to evaluate.
+
+        Returns
+        -------
+        causal_score_dict
+            Dictionary of scores obtain for different perturbations
```

Comment:
While reading this it is very unclear.. Is it for different pertubations ? Or is it the same pertubation with an incremental process ?

Reply:
It is the same perturbations, the computation is just divided so that an user can have access to all computed values. It is useful to visualize insertion and deletion curves.
Further precisions in the documentation will be added